test(napi/oxlint): add tests for without JS plugins enabled

overlookmotel opened the pull request:
Add a couple of tests to make sure `napi/oxlint` operates correctly without `--experimental-js-plugins` CLI option.

camc314 reviewed (approved):
should we add some tests for the error flows (low priority)

i.e. what happens when you run oxlint without `--experimental...` but yoy have plugins configured

overlookmotel committed:
test(napi/oxlint): add tests for without JS plugins enabled (#13698)

Add a couple of tests to make sure `napi/oxlint` operates correctly without `--experimental-js-plugins` CLI option.
